openid-4-verifiable-presentations-1_0.md

```diff
@@ -329,6 +329,24 @@ The following is a non-normative example of a transaction data content, after ba
 }
 ```
 
+`attachments`:
+: OPTIONAL. Array of objects, where each object includes an extra resource that needs to be passed to the Wallet. These resources are intended to provide additional context to the Wallet or the End-User. The Wallet MUST ignore any unrecognized attachment formats. Each object contains the following parameters:
+
+    * `format`: REQUIRED: String that identifies the format of the attachment.
+    * `data`: REQUIRED: The structure and content of data are format-dependent and not constrained by this specification.
+    * `credential_ids`: OPTIONAL. Array of strings each referencing a Credential requested by the Verifier for which the attachment is relevant. Each string matches the `id` field in a DCQL Credential Query. If there is more than one element in the array, the Wallet MUST use only one of the referenced Credentials for transaction authorization.
```

Comment:
> Array of strings each referencing a Credential requested by the Verifier for which the attachment is relevant

It feels like we should probably say something about what the meaning of omitting `credential_ids` is? I guess something like "If omitted the attachment is relevant to all requested credentials."

Reply:
I updated it to make clear that the usage depends on the `format` value and therefore the context how to use it.

Reply:
> Or not at all. You could also pass something like a CRL list or so that has nothing to do directly with the requested credentials, but is needed to process the request (in this case validating the status of the x509 that was used for signing).

I think I feel nervous about this. It feels like we're defining an extension point on top of something that's already an extension point. If you need to pass a CRL list why wouldn't it be better to define an OAuth level parameter for passing the CRL in? (And profiles can absolutely define extra ecosystem-specific oauth parameters.)

Reply:
I also thought about this, so I only gave examples that are relevant in the credentials context like the requirements from the EU (registration certificates and authorization attestations).

We can also write to limit it that you only need attachments there that are required to fetch credentials.